### ISSUE TITLE
ConnectionError: RemoteDisconnected - Remote end closed connection without response

### ISSUE THREAD

**Issue by anyu** [box/box-python-sdk#756]
- [X] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [X] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [X] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [X] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue

Our app sends a consistent volume of requests at regular intervals to the [GET folder information](https://developer.box.com/reference/get-folders-id/) endpoint.

Over the last 2 months at least, we've started seeing that a handful fail in between successful requests due to the `RemoteDisconnected` error below. The only notable pattern identified so far has been that there seems to be an uptick in these errors around mid-month through the end of the month, despite consistent request volume.

If we're hitting any sort of rate limit, we'd expect a 429, which we do get for other requests.

Are there any additional limits from Box's end that would result in such errors, and not be 429s?

The docs on [Rate Limits](https://developer.box.com/guides/api-calls/permissions-and-errors/rate-limits/#licensing-based) has a section that says 

>  If Box's infrastructure detects that a tool used by or on behalf of a customer has exceeded that customer's API license allocation or is intending to circumvent network controls, **additional selective rate-limiting** may be imposed. 

Would these 'additional selective rate-limiting measures' return 429s?

### Error Message, Including Stack Trace

```
["\u003cclass 'requests.exceptions.ConnectionError'\u003e","('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))","\u003ctraceback object at 0x7f74d558e300\u003e"]
```

### Versions Used
**Python SDK:** boxsdk 3.3.0
**Python:** 3.9.13

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-python-sdk/search?type=Issues

**Comment by mwwoda** [comment]
Hi @anyu 
It looks like your connection was forcibly closed before receiving a response. This could be a server issue, or it could happen if your environment has changed between requests (such as network issues). 

I don't think we can easily reproduce this issue in sdk because of its non-deterministic nature and it looks more like a server problem than a sdk problem. My advice is to contact Box Support https://support.box.com/hc/en-us/requests/new so that this problem can be further investigated on the backend side.

**Comment by anyu** [comment]
We've chatted with Box Support and unfortunately a cause hasn't been identified.

**Comment by lukaszsocha2** [comment]
I guess with the fix released in pythonsdk 3.5.1 it won't be necessary any more. You should be able to cache and reuse one client object now.
